examples/llm_qlora_data_parallel/train.py

```diff
@@ -0,0 +1,164 @@
+import logging
+import os
+
+import numpy as np
+import pandas as pd
+import yaml
+
+from ludwig.api import LudwigModel
+
+np.random.seed(123)
+
+
+# Llama-2-7b-hf requires HUGGING_FACE_HUB_TOKEN to be set as an environment variable
+# You can get a token at https://huggingface.co/settings/tokens
+if "HUGGING_FACE_HUB_TOKEN" not in os.environ:
+    raise ValueError(
+        "Please set your Hugging Face Hub token as an environment variable using `export "
+        "HUGGING_FACE_HUB_TOKEN=your_token`. You can get a token at https://huggingface.co/settings/tokens"
+    )
+
+df = pd.read_json("https://raw.githubusercontent.com/sahil280114/codealpaca/master/data/code_alpaca_20k.json")
+
+# We're going to create a new column called `split` where:
+# 90% will be assigned a value of 0 -> train set
+# 5% will be assigned a value of 1 -> validation set
+# 5% will be assigned a value of 2 -> test set
+# Calculate the number of rows for each split value
+total_rows = len(df)
+split_0_count = int(total_rows * 0.9)
+split_1_count = int(total_rows * 0.05)
+split_2_count = total_rows - split_0_count - split_1_count
+
+# Create an array with split values based on the counts
+split_values = np.concatenate([np.zeros(split_0_count), np.ones(split_1_count), np.full(split_2_count, 2)])
+
+# Shuffle the array to ensure randomness
+np.random.shuffle(split_values)
+
+# Add the 'split' column to the DataFrame
+df["split"] = split_values
+df["split"] = df["split"].astype(int)
+
+# For now, just use 1000 rows as a demonstration
+df = df.head(n=1000)
```

Comment:
Instead of doing `.head()`, we can use @Infernaught's `preprocessing.sample_size` parameter.

Reply:
@Infernaught Can we document this in Ludwig docs as well?